numpy/testing/_private/utils.py

```diff
@@ -821,7 +821,12 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
 
                     # note: this definition of relative error matches that one
                     # used by assert_allclose (found in np.isclose)
-                    max_rel_error = (error / abs(y)).max()
+                    # Filter values where the divisor would be zero
+                    nonzero = y != 0
+                    if all(~nonzero):
```

Comment:
`nonzero` is guaranteed to be an array here I think, so it would be better to use `nonzero.all()` (the other one may be pythons `all`). Other than that, seems all fine to me.

Reply:
```nonzero``` isn't necessarily an array, for example in test ```TestArrayEqual.test_subclass_that_overrides_eq``` the not equal operator returns a single bool - calling ```.all()``` on it would raise an ```AttributeError```. You're right about ```all``` being python's ```all``` - I'll import the function from numpy to fix that.

Also, applying operator ```~``` to a bool wouldn't work as intended, so I think it would be better to use ```any```:
```python
nonzero = y != 0
if not any(nonzero):
	max_rel_error = array(inf)
else:
	max_rel_error = (error[nonzero] / abs(y[nonzero])).max()
```
That would work even if the not equal operator returns a single bool, but it's kind of hard to read, so we could also invert the if statement.
Thoughts?

Reply:
A numpy bool woul dbe good enough and considering that indexing works below it should do. But `np.all` is fine as well.

Reply:
So, simply this?
```python
nonzero = bool_(y != 0)
if all(~nonzero):
	max_rel_error = array(inf)
else:
	max_rel_error = (error[nonzero] / abs(y[nonzero])).max()
```

Reply:
Just make it `np.any` and I will be happy (or tell me that it already uses `np.any`).

Reply:
Do you mean ```np.all``` here? Yes, ```all``` is imported at the top of the function.